lib/src/run_interactive_codemod.dart

```diff
@@ -141,28 +139,11 @@ Future<int> runInteractiveCodemodSequence(
       }
       return ExitCode.success.code;
     }
-    final doResolve = suggestors.any((s) => s is ResolvedSuggestor);
-    final resolved = <String, CompilationUnit>{};
-    if (doResolve) {
-      stderr.writeln('Analyzing...');
-      final collection = AnalysisContextCollection(
-          includedPaths: filePaths.map((p) => path.canonicalize(p)).toList());
-      for (final file in filePaths) {
-        final f = path.canonicalize(file);
-        final context = collection.contextFor(f);
-        final unit = await context.currentSession.getResolvedUnit(f);
-        resolved[file] = unit.unit;
```

Comment:
Instead of doing all of the analysis upfront, the `FileContext` wrapper allows us to defer that work until a suggestor actually requests the information for a given file. Multiple suggestors operating on the same files should be sharing the same analysis context, which means the analysis work should be shared, as well.